framework/packages/abstract-interface/src/native/account_factory.rs

```diff
@@ -77,17 +77,30 @@ impl<Chain: CwEnv> AccountFactory<Chain> {
             funds,
         )?;
 
+        // Parse data from events
+        let acc_seq = &result.event_attr_value(ABSTRACT_EVENT_TYPE, "account_sequence")?;
+        let trace = &result.event_attr_value(ABSTRACT_EVENT_TYPE, "trace")?;
+        let id = AccountId::new(
+            acc_seq.parse().unwrap(),
+            AccountTrace::try_from((*trace).as_str())?,
+        )?;
+        // construct manager and proxy ids
+        let manager_id = format!("{MANAGER}-{}", id.to_string());
+        let proxy_id = format!("{PROXY}-{}", id.to_string());
+
+        // set addresses
         let manager_address = &result.event_attr_value(ABSTRACT_EVENT_TYPE, "manager_address")?;
         self.get_chain()
             .state()
-            .set_address(MANAGER, &Addr::unchecked(manager_address));
+            .set_address(&manager_id, &Addr::unchecked(manager_address));
         let proxy_address = &result.event_attr_value(ABSTRACT_EVENT_TYPE, "proxy_address")?;
         self.get_chain()
             .state()
             .set_address(PROXY, &Addr::unchecked(proxy_address));
```

Comment:
That is really nice! set_address for proxy should be &proxy_id as well, I suppose?